### ISSUE TITLE
Graphql with dataloader doesn't return the list, only first item

### ISSUE THREAD

**Issue by jefflequeux** [w3tecch/express-typescript-boilerplate#59]
When I change a userId of a pet so that one user will have 2 pets:

| id | name | age | user_id |
| ------------- |:-------------:|:-------------:| -----:|
|'0e5d9edb-56d4-43df-b96d-b186dcf8a085' | 'Darius' | '96328' | '16919324-89c1-447e-a770-134665993acc' |
|'21c8cc56-0e71-4ef7-b373-707ec0a7fe97' | 'Jadyn' | '27435' | '16919324-89c1-447e-a770-134665993acc' |

```

export const UserType = new GraphQLObjectType({
    name: 'User',
    description: 'A single user.',
    fields: () => ({ ...UserFields, ...{
        pets: {
            type: new GraphQLList(PetOfUserType),
            description: 'The pets of a user',
            resolve: async (user: User, args: any, context: GraphQLContext<any, any>) =>
                // We use data-loaders to save db queries
                context.dataLoaders.petByUserIds.loadMany([user.id]),
                // This would be the case with a normal service, but not very fast
                // context.container.get<PetService>(PetService).findByUser(user),
        },
    } }),
});
```
Will return only **1** pet
but:
```
export const UserType = new GraphQLObjectType({
    name: 'User',
    description: 'A single user.',
    fields: () => ({ ...UserFields, ...{
        pets: {
            type: new GraphQLList(PetOfUserType),
            description: 'The pets of a user',
            resolve: async (user: User, args: any, context: GraphQLContext<any, any>) =>
                // We use data-loaders to save db queries
                // context.dataLoaders.petByUserIds.loadMany([user.id]),
                // This would be the case with a normal service, but not very fast
                context.container.get<PetService>(PetService).findByUser(user),
        },
    } }),
});
```
Will return **2** pets.

**Comment by jefflequeux** [comment]
I was able to fix it by replacing **createDataLoader**:

```
export function createDataLoader<T>(obj: ObjectType<T>, method?: string, key?: string): any {
    let repository;
    try {
        repository = getCustomRepository<Repository<any>>(obj);
    } catch (errorRepo) {
        try {
            repository = getRepository(obj);
        } catch (errorModel) {
            throw new Error('Could not create a dataloader, because obj is nether model or repository!');
        }
    }
    return new DataLoader<number, any>(async (ids: number[]) => {
        let items = [];
        if (method) {
            items = await repository[method](ids);
        } else {
            items = await repository.findByIds(ids);
        }
        const gs = _.groupBy(items, key);
        return ids.map(id => gs[id] || []);
    });
}
```

**Comment by hirsch88** [comment]
Hi @jefflequeux 
Thanks for your input. I will check that and add it to the next release
Cheers,
hirsch88

**Comment by hirsch88** [comment]
Hi @jefflequeux 
This issue is fixed in the new release 3.0.0. Thanks for reporting.
Cheers,
hirsch88
